peer/brontide.go

```diff
@@ -634,7 +634,35 @@ func (p *Brontide) Start() error {
 	go p.writeHandler()
 	go p.readHandler()
 	go p.channelManager()
-	go p.pingHandler()
+
+	var (
+		lastBlockHeader           *wire.BlockHeader
+		lastSerializedBlockHeader [wire.MaxBlockHeaderPayload]byte
+	)
+	newPingPayload := func() []byte {
+		// Each time a new block comes in, we'll copy the raw header
```

Comment:
> I think it is worth having this as its own method rather than polluting the Start method with this logic.

Maybe, though it's not quite that simple with the variable capture for the closure, but I can solve that by making a small subsystem and put those two vars in a struct that is the receiver of said method.